### ISSUE TITLE
CopyInBucket returns wrong error 

### ISSUE THREAD

**Issue by mvrhov** [minio/minio-go#1461]
When you use CopyInBucket functionality and source object doesn't exist, the returned error reports NoSuckKeyError, where the Key property contains the destination object. Which doesn't make any sense.
I'd say that the key property should contain source object.

**Comment by vadmeste** [comment]
> I've changed the title, as I have used the wrong function name. The proper minio api function is CopyObject

Thanks.

I tested CopyObject() code and it looks it is returning correct source key name when the latter doesn't exist. Could you paste a simple reproducer of this ?

**Comment by mvrhov** [comment]
Modified CopyObject example... https://play.golang.org/p/wnJpUpbbBrf. The output should be: 
_The specified key does not exist., **my-sourceobjectname**_

**Comment by vadmeste** [comment]
> Modified CopyObject example... https://play.golang.org/p/wnJpUpbbBrf. The output should be:
> _The specified key does not exist., **my-sourceobjectname**_

yes thanks. It looks like this is a MinIO server bug, not minio-go. I will send a fix.
